### PR TITLE
feat: Add support for extensible attributes

### PR DESCRIPTION
**Pull request by JoooostB**
Some Infoblox environments require extensible attributes to be set like `Tenant ID`, `Cloud API Owned` and `CMP Type`. (See [Infoblox docs](https://docs.infoblox.com/space/nios90/280407399/Extensible+Attributes+for+Cloud+Objects) for available default available attributes.)

This PR adds the ability to set these attributes from JSON using environment variable `INFOBLOX_EXTENSIBLE_ATTRIBUTES_JSON`:
```ini
...
INFOBLOX_EXTENSIBLE_ATTRIBUTES_JSON="{\"CMP Type\":\"external-dns\",\"Cloud API Owned\":\"True\",\"Tenant ID\":\"1337\"}"
...
```

### CODE REVIEW

**Review by johnvanhienen — APPROVED** [review]

File: internal/infoblox/infoblox.go [hunk 1]
```diff
@@ -784,3 +803,22 @@ func lookupEnvAtoi(key string, fallback int) (i int) {
...
+			// Input was a stringified JSON, unmarshal it again
+			extAttrJSON = intermediate
+		}
+		if err := json.Unmarshal([]byte(extAttrJSON), &extAttrs); err != nil {
+			return nil, fmt.Errorf("cannot process 'ext_attrs' field: %w", err)
+		}
+	}
+	if extAttrs == nil {
+		extAttrs = make(map[string]interface{})
+	}
```
Comment (kuritka) [inline internal/infoblox/infoblox.go:822]: can this ever happen?

File: internal/infoblox/infoblox.go [hunk 2]
```diff
@@ -784,3 +803,22 @@ func lookupEnvAtoi(key string, fallback int) (i int) {
 	}
 	return
 }
+
+func deserializeEAs(extAttrJSON string) (map[string]interface{}, error) {
+	extAttrs := make(map[string]interface{})
+	if extAttrJSON != "" {
```
Comment (kuritka) [inline internal/infoblox/infoblox.go:809]: consider invert condition

File: internal/infoblox/infoblox.go [hunk 3]
```diff
@@ -784,3 +803,22 @@ func lookupEnvAtoi(key string, fallback int) (i int) {
...
+		// Check if the input is stringified JSON
+		var intermediate string
+		if err := json.Unmarshal([]byte(extAttrJSON), &intermediate); err == nil {
```
Comment (kuritka) [inline internal/infoblox/infoblox.go:812]: code continues even if error occures, is it ok ?

**Review by kuritka — CHANGES_REQUESTED** [review]
Hi @JoooostB, thanks a ton for your input and I'll be glad if we can merge it. Can you fix the JSON processing ? (if you have the desire and capacity, feel free to add a unit test with corner cases, but it is not necessary).

🇳🇱

File: internal/infoblox/infoblox.go (hunk 3 above)
Comment (JoooostB) [reply internal/infoblox/infoblox.go:812]: I agree this may not be the cleanest of solutions, but I fail to come up with a neater alternative. 

If it successfully unmarshals `extAttrJSON` into a string, it means `extAttrJSON` was a stringified JSON, and should be marshalled again (hence the `extAttrJSON = intermediate` declaration).
If not, it proceeds to unmarshal `extAttrJSON` as a normal JSON object into `extAttrs`. 

Do you happen to see a better alternative on approaching this situation?

File: internal/infoblox/infoblox.go [hunk 4]
```diff
@@ -784,3 +803,20 @@ func lookupEnvAtoi(key string, fallback int) (i int) {
...
+func deserializeEAs(extAttrJSON string) (map[string]interface{}, error) {
+	extAttrs := make(map[string]interface{})
+	if extAttrJSON == "" {
+		return extAttrs, nil
+	}
+	// Check if the input is stringified JSON
+	var intermediate string
+	if err := json.Unmarshal([]byte(extAttrJSON), &intermediate); err == nil {
+		extAttrJSON = intermediate
+	}
```
Comment (kuritka) [inline internal/infoblox/infoblox.go:816]: thx for change, its more readable now. Isn't this section also redundant ? If the input is not valid, you can find out about it directly in the following section. My point is that an invalid input will throw the exact same error in the next section as it would in this section. If the entry is valid, then the next section will be valid too.

Comment (JoooostB) [reply internal/infoblox/infoblox.go:816]: You're right, this was a "fix" for a problem that doesn't exist but is caused by user error (myself included, I misconfigured my input JSON....)

**Review by kuritka — APPROVED** [review]
thx for your effor @JoooostB 🚀  and don't forget to give us the ⭐
